cmd/proxy/actions/storage.go

```diff
@@ -50,6 +51,14 @@ func GetStorage(storageType string, storageConfig *config.StorageConfig) (storag
 			return nil, errors.E(op, "Invalid CDN Storage Configuration")
 		}
 		return gcp.New(context.Background(), storageConfig.GCP, storageConfig.CDN)
+	case "s3":
+		if storageConfig.S3 == nil {
+			return nil, errors.E(op, "Invalid S3 Storage Configuration")
+		}
+		if storageConfig.CDN == nil {
```

Comment:
i'm wondering why are we failing on nil,
we use CDN config as this 
```go
func (s *Storage) BaseURL() *url.URL {
	return s.cdnConf.CDNEndpointWithDefault(s.baseURI)
}
```
if we move this nil check to BaseURL
```gofunc (s *Storage) BaseURL() *url.URL {
	if(s.cdnConf == nil){
		return s.baseUri
	}
	return s.cdnConf.CDNEndpointWithDefault(s.baseURI)
}
```

it should be also fine, and allows me to shrink my config file a bit

thought? @t-tomalak @rchakra3

Reply:
Droped that dependency and now it's not required